src/app/frontend/replicasetdetail/replicasetdetail_controller.js

```diff
@@ -156,6 +165,19 @@ export default class ReplicaSetDetailController {
    */
 
   /**
+   * Updates replicas count in replica set
+   * @param {number} replicasCount
+   * @param {function(!backendApi.ReplicaSetDetail)=} opt_callback
+   * @param {function(!angular.$http.Response)=} opt_errback
+   * @private
+   */
+  updateReplicas_(replicasCount, opt_callback, opt_errback) {
+    this.replicaSetDetail.podsDesired = replicasCount;
```

Comment:
Only properties that have changed should be sent back to the server. There is many reasons to do so. First, if you send entire object you can have race conditions with other clients that already modify this object (e.g., imagine that in another tab you changed the name of the replica set). Other thing is that backend method needs to be generic enough to diff against all properties, which is harder to implement and read than to write specialized methods for concrete use cases.